.travis.yml

```diff
@@ -12,13 +12,9 @@ env:
   - OPAMVERBOSE="true"
   - TEST_TARGETS="unit_tests real_tests"
   matrix:
-  - OCAML_VERSION=4.04.1
```

Comment:
For what reason should we stop testing with early versions of OCaml?

Reply:
I used `Seq` somewhere

Reply:
and all Pervasives data structures now converts to and from `Seq`

Reply:
I think that we should support as many versions as possible. Standard data structures already have their own (optimized) iterators.

Reply:
@rlepigre, should this conversation be closed?

Reply:
@fblanqui: the problem is still here, and I think it should be fixed.

Reply:
I can review the use of Seq, even though, IMO, it looks like a good practice.  But I use the [choose_opt] for maps and sets quite a lot, which are available from ocaml 4.06.

Reply:
In other comments, I was wondering whether Seq was really more efficient. It would be nice to have some evidence indeed.

Reply:
It appears handy if, for instance, one want to convert an array to a list and apply a function on the elements; one can do `Array.to_seq a |> Seq.map f |> Seq.to_list`, which should--if I'm not mistaken--traverse the array `a` only once.

Reply:
See [here](https://ocamlverse.github.io/content/optimizing_performance.html) and [there](https://ocamlverse.github.io/content/iterators.html)

Reply:
I see. That's nice! For the record, thanks to add a comment in the code with a link to https://ocamlverse.github.io/content/iterators.html . Thanks to add a sentence and a link in doc/sections/implementation.md too.

Reply:
I'm not at all convinced with the argument that this makes code more efficient. For me, using iterators is just one more obfuscation layer (pretty much like `|>` and similar operators). In my opinion, it is very important to keep the code as simple as possible, and using every new fancy feature of the standard library (and the language in general) does the opposite way.

@fblanqui: I'll let you decide where you want to go with that, but I definitely disagree with the way things are going here.

Reply:
@gabrielhdt , would it be possible to implement a version without Seq and see the difference on the libraries and your generated examples?

Reply:
Perhaps, it suffices to provide an implementation of Seq based on List.

Reply:
Yes I can try

Reply:
In fine I don't use Seq anymore.  However `List.init` requires ocaml >= 4.06.

Reply:
If it's only `List.init`, I think you could just define a recursive function on the site of the call. That may even simplify what you are doing. From what I understand to your use of the function, it would be faster to actually do so since you could build the list from the end, and hence be tailrec without using a reversal (as I'm guessing `List.init` does).

Reply:
There are the `.*_opt` operations as well, which are available from 4.05

Reply:
Well, it's not like these functions are absolutely necessary for what you are doing. Just use their non-opt versions (which I prefer), or define them in the `Extra` module if you cannot live without them.

Reply:
I don't see the point in writing in lambdapi what caml already offers.

Reply:
It is good practice to support many versions of OCaml, especially when the effort is minimal. For instance `coq` only requires `ocaml 4.02.3`.

And you would not need to write extra code if you used the non-opt versions. Exceptions are a core design choice of OCaml and its theory, and I really don't see why they should be avoided.

Reply:
I see.  I'll let @fblanqui decide which direction we take.

Reply:
Thanks to follow Rodolphe's advice.